### PR TITLE
Memo Modal

### PR DESCRIPTION
**Pull request by tharmaman**

### CODE REVIEW

File: src/foam/nanos/approval/ApprovalRequest.js [hunk 1]
```diff
@@ -46,7 +47,8 @@
     'ctrl',
     'currentMenu',
     'stack',
-    'user'
+    'user',
```
Comment (jlhughes) [inline src/foam/nanos/approval/ApprovalRequest.js]: 'subject' please, unless some other part of approval is exporting 'user'.

File: src/foam/nanos/approval/ApprovalRequest.js [hunk 2]
```diff
@@ -527,26 +582,11 @@
...
-          this.throwError.pub(e);
-          X.ctrl.add(this.NotificationMessage.create({
-            message: e.message,
-            type: 'error'
-          }));
-        });
+        this.objectSummaryView.add(this.Popup.create({ backgroundColor: 'transparent' }).tag({
+          class: "foam.u2.MemoModal",
+          onExecute: this.approveWithMemo.bind(this),
+          title: "Add a memo to this (optional)"
```
Comment (jlhughes) [inline src/foam/nanos/approval/ApprovalRequest.js]: This does not read well.  Perhaps
'Please provide a reason'

File: src/foam/nanos/approval/ApprovalRequest.js [hunk 3]
```diff
@@ -563,26 +603,12 @@
...
-          X.ctrl.add(this.NotificationMessage.create({
-            message: e.message,
-            type: 'error'
-          }));
-        });
+        this.objectSummaryView.add(this.Popup.create({ backgroundColor: 'transparent' }).tag({
+          class: "foam.u2.MemoModal",
+          onExecute: this.rejectWithMemo.bind(this),
+          isMemoRequired: true,
+          title: "Add a memo to this (required)"
```
Comment (jlhughes) [inline src/foam/nanos/approval/ApprovalRequest.js]: This does not read well.  Perhaps
'Please provide a reason'

File: src/foam/u2/MemoModal.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    {
+      class: 'Function',
+      name: 'onExecute'
+    },
+    {
+      class: 'String',
+      name: 'title',
+      expression: isMemoRequired => {
+        if (isMemoRequired)
+          return "Add a memo to this (required)";
```
Comment (jlhughes) [inline src/foam/u2/MemoModal.js]: Perhaps 'Please add a note'

File: src/foam/nanos/approval/ApprovalRequest.js (hunk 1 above)
Comment (tharmaman) [reply src/foam/nanos/approval/ApprovalRequest.js]: Removed. Wasn't even used in the first place.

File: src/foam/nanos/approval/ApprovalRequest.js (hunk 2 above)
Comment (tharmaman) [reply src/foam/nanos/approval/ApprovalRequest.js]: Done.

File: src/foam/nanos/approval/ApprovalRequest.js (hunk 3 above)
Comment (tharmaman) [reply src/foam/nanos/approval/ApprovalRequest.js]: Done.

File: src/foam/u2/MemoModal.js (hunk 4 above)
Comment (tharmaman) [reply src/foam/u2/MemoModal.js]: Done.

**Review by jlhughes — APPROVED** [review]

**Pull request merged**